sidekick_core/lib/src/template/sidekick_package.template.dart

```diff
@@ -0,0 +1,269 @@
+import 'package:recase/recase.dart';
+
+import 'package:sidekick_core/sidekick_core.dart';
+import 'package:sidekick_core/src/template/download_dart.sh.template.dart';
+import 'package:sidekick_core/src/template/entrypoint.template.dart';
+import 'package:sidekick_core/src/template/install.sh.template.dart';
+import 'package:sidekick_core/src/template/run.sh.template.dart';
+import 'package:sidekick_core/src/template/sidekick_config.sh.template.dart';
+
+class SidekickTemplate {
+  void generate(SidekickTemplateProperties props) {
+    generateEntrypoint(props);
+    generatePackage(props);
+    generateTools(props);
+  }
+
+  void generateEntrypoint(SidekickTemplateProperties props) {
+    final path = relative(
+      props.packageLocation.path,
+      from: props.entrypointLocation.parent.path,
+    );
+    final entrypoint = entrypointTemplate(packagePath: path);
+    props.entrypointLocation.writeAsStringSync(entrypoint);
+  }
+
+  void generateTools(SidekickTemplateProperties props) {
```

Comment:
The separate methods will allow us to reuse the template for the update mechanism